mesonbuild/build.py

```diff
@@ -1301,6 +1301,16 @@ def add_deps(self, deps):
         for dep in deps:
             if dep in self.added_deps:
                 continue
+
+            dep_d_features = dep.get_d_features()
+
+            for feat in ('versions', 'import_dirs'):
+                if feat in dep_d_features:
+                    if feat in self.d_features:
+                        self.d_features[feat].extend(dep_d_features[feat])
+                    else:
+                        self.d_features[feat] = dep_d_features[feat]
```

Comment:
We can simplify this if we use a `collections.defaultdict` here:
```suggestion
            for feat in ('versions', 'import_dirs'):
                if feat in dep_d_features:
                    self.d_features[feat].extend(dep_d_features[feat])
```